sklearn/impute.py

```diff
@@ -115,16 +137,37 @@ class SimpleImputer(BaseEstimator, TransformerMixin):
     Notes
     -----
     Columns which only contained missing values at `fit` are discarded upon
-    `transform`.
+    `transform` if strategy is not "constant".
 
     """
-    def __init__(self, missing_values="NaN", strategy="mean",
-                 verbose=0, copy=True):
+    def __init__(self, missing_values=np.nan, strategy="mean",
+                 fill_value=None, verbose=0, copy=True):
         self.missing_values = missing_values
         self.strategy = strategy
+        self.fill_value = fill_value
         self.verbose = verbose
         self.copy = copy
 
+    def _validate_input(self, X):
+        allowed_strategies = ["mean", "median", "most_frequent", "constant"]
+        if self.strategy not in allowed_strategies:
+            raise ValueError("Can only use these strategies: {0} "
+                             " got strategy={1}".format(allowed_strategies,
+                                                        self.strategy))
+
+        if self.strategy in ("most_frequent", "constant"):
+            dtype = None
+        else:
+            dtype = FLOAT_DTYPES
+
+        if self.missing_values is not np.nan:
```

Comment:
we still need to support 'NaN', but issue a deprecation warning that it will not be supported (and will be treated as a constant string value) from version 0.22.

Reply:
Actually, the SimpleImputer is new in 0.20, so we decided to use np.nan as default value instead of 'NaN' without the need of deprecation. We didn't touch the default value of the processing.Imputer which will be deprecated